### PR TITLE
refactor(plugins)!: remove nvim-notify from core

### LINKED ISSUE #3294: remove nvim-notify from the core plugins
Unfortunately, [nvim-notify](https://github.com/rcarriga/nvim-notify) has been causing way too many issues and instabilities.

several plugins are now using `vim.notify` notice that would trigger in the middle of packer's operations, at that point `nvim-notify` lost its state or something, and you get crazy errors like this

<details>
<summary>expand</summary>

```gdb
Error executing vim.schedule lua callback: ...ker/start/nvim-notify/lua/notify/serv
ice/buffer/init.lua:98: Vim(set):E474: Invalid argument: colorcolumn=value
stack traceback:
        [C]: in function 'nvim_buf_set_option'
        ...ker/start/nvim-notify/lua/notify/service/buffer/init.lua:98: in function 'render'
        ...ack/packer/start/nvim-notify/lua/notify/service/init.lua:53: in function 'push'
        ...m/site/pack/packer/start/nvim-notify/lua/notify/init.lua:209: in function 'impl'
        ...start/structlog.nvim/lua/structlog/sinks/nvim_notify.lua:49: in function 'write'
        ...ack/packer/start/structlog.nvim/lua/structlog/logger.lua:58: in function 'log'
        ...ack/packer/start/structlog.nvim/lua/structlog/logger.lua:64: in function 'log'
        ...cedraz-/.local/share/lunarvim/lvim/lua/lvim/core/log.lua:155: in function 'add_entry'
        ...cedraz-/.local/share/lunarvim/lvim/lua/lvim/core/log.lua:198: in function 'info'
        ...raz-/.local/share/lunarvim/lvim/lua/lvim/utils/hooks.lua:22: in function 'run_on_packer_comp
lete'
        ...z-/.local/share/lunarvim/lvim/lua/lvim/plugin-loader.lua:50: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>

```

</details>

it feels like so many plugins are spamming `vim.notify` to the point where it's hard to keep track of all of these popups.. :disappointed:  

also, we've had to create sketchy workarounds just for the sake of having this plugin

<details>
<summary>expand</summary>

https://github.com/LunarVim/LunarVim/blob/c015d0feeed35b34d84558a8e4a49549b3b67781/lua/lvim/core/nvimtree.lua#L155-L166

</details>

here are some related issues and complaints I've managed to find
- https://github.com/LunarVim/LunarVim/issues/3265
- https://github.com/LunarVim/LunarVim/issues/2713
- https://github.com/LunarVim/LunarVim/issues/2242
- https://github.com/LunarVim/LunarVim/issues/1821
- https://github.com/LunarVim/LunarVim/issues/2778
- https://github.com/LunarVim/LunarVim/issues/2960
- https://github.com/LunarVim/LunarVim/issues/2734
- https://github.com/LunarVim/LunarVim/issues/2738
- https://github.com/LunarVim/LunarVim/issues/2673

### PR DESCRIPTION
**Pull request by mhuggins7278**
# Description

removing nvim-notify 

fixes #3294

## How Has This Been Tested?

- Run command `:mycommand`
- Check logs
Tested locally 
The only issue i ran into is as follows. 

I had the following line in my config
```lua
lvim.buitlin.notify.enable = true
```
which resulted in the following error
2022-10-19 21:21:47 [WARN ] lvim: "Invalid configuration: /Users/MHuggins/.config/lvim/lua/user/options.lua:30: attempt to index field 'notify' (a nil value)" file="lua/lvim/config/init.lua", line=119
 

simply removing the line from my config resolved the issue not sure if there is a better way to handle that

### CODE REVIEW

**Comment by kylo252** [comment]
> simply removing the line from my config resolved the issue not sure if there is a better way to handle that

we have a deprecation handler in `lvim/config/init.lua` that we need to use. 

I'm not sure if it's necessary to create a meta-table to handle nested options 🤔

@abzcoding, wdyt?

**Comment by abzcoding** [comment]
> > simply removing the line from my config resolved the issue not sure if there is a better way to handle that
> 
> we have a deprecation handler in `lvim/config/init.lua` that we need to use.
> 
> I'm not sure if it's necessary to create a meta-table to handle nested options 🤔
> 
> @abzcoding, wdyt?

agreed, we should alert users about the deprecation of `nvim-notify`

**Comment by mhuggins7278** [comment]
Ok I'll take a look and update the PR

**Comment by mhuggins7278** [comment]
I've added the deprecation notice 
image

File: lua/lvim/core/nvimtree.lua [hunk 1]
```diff
@@ -147,24 +147,11 @@ function M.setup()
     return
   end
 
-  local status_ok_1, utils = pcall(require, "nvim-tree.utils")
+  local status_ok_1 = pcall(require, "nvim-tree.utils")
   if not status_ok_1 then
     return
   end
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:153]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ All notable changes to this project will be documented in this file.
 
 ### <!-- 1 --> Features
 
+- _(notify)_ remove nvim.notify ([#3300](https://github.com/LunarVim/LunarVim/pull/3300))
```
Comment (kylo252) [inline CHANGELOG.md:9]:
```suggestion
```

you can skip this for now :)

File: lua/lvim/core/nvimtree.lua (hunk 1 above)
Comment (mhuggins7278) [reply lua/lvim/core/nvimtree.lua]: cool ty.. not sure why i just didn't remove the whole function initially :-(

**Review by ChristianChiarulli — APPROVED** [review]

**Comment by ChristianChiarulli** [comment]
LGTM

**Comment by danielo515** [comment]
Why not just deactivate by default so it is easier to activate for those who want?

**Comment by kylo252** [comment]
> Why not just deactivate by default so it is easier to activate for those who want?

because that means that lunarvim is _still_ maintaining compatibility, which is precisely what we want to avoid here. 

If it works, great! otherwise, it's easier for users to see the ~~problems~~ caveats with using this plugin, and then they're free to set whatever options fits their needs.

**Comment by latipun7** [comment]
There's a leftover here:
https://github.com/LunarVim/LunarVim/blob/25c4241b07d2036fc5d0d4c18be43a16a5370ef0/lua/lvim/core/which-key.lua#L232

**Pull request merged**